core/src/main/java/org/apache/calcite/sql/type/SqlTypeFactoryImpl.java

```diff
@@ -88,6 +90,15 @@ public SqlTypeFactoryImpl(RelDataTypeSystem typeSystem) {
     if (maxPrecision >= 0 && precision > maxPrecision) {
       precision = maxPrecision;
     }
+    if (typeName == SqlTypeName.DECIMAL) {
+      if (precision != RelDataType.PRECISION_NOT_SPECIFIED && precision <= 0) {
+        throw RESOURCE.invalidPrecisionForDecimalType(precision, maxPrecision).ex();
+      }
+      if (scale != RelDataType.SCALE_NOT_SPECIFIED && scale < 0
```

Comment:
This is now a problem, since SCALE_NOT_SPECIFIED is -1, which may be a legal scale value. Maybe this constant should be changed to Integer.MIN_VALUE.

Reply:
SCALE_NOT_SPECIFIED is Integer.MIN_VALUE.  PRECISION_NOT_SPECIFIED is -1.